### ISSUE TITLE
module 'asyncio' has no attribute 'TaskGroup'

### ISSUE THREAD

**Issue by jimemo** [sbtinstruments/aiomqtt#267]
python3 -V

`Python 3.9.2`

Run exact code from: https://sbtinstruments.github.io/aiomqtt/subscribing-to-a-topic.html#processing-concurrently

```
Exception has occurred: AttributeError
module 'asyncio' has no attribute 'TaskGroup'
  File "/opt/bitnami/apache2/htdocs/py/mqtt_mysql/test_aiomqtt.py", line 14, in main
    async with asyncio.TaskGroup() as tg:
  File "/opt/bitnami/apache2/htdocs/py/mqtt_mysql/test_aiomqtt.py", line 19, in <module>
    asyncio.run(main())
AttributeError: module 'asyncio' has no attribute 'TaskGroup'
```

I'm not able to change Python version at the moment. Is there a bug fix available or an aiomqtt upgrade or downgrade option available?

**Comment by JonathanPlasse** [comment]
`asyncio.TaskGroup` is only available in Python 3.11+.
You could use [AnyIO](https://anyio.readthedocs.io/en/stable/tasks.html).
AnyIO use asyncio under the hood and makes using async in Python much easier.

**Comment by JonathanPlasse** [comment]
This is the equivalent code.
```python
import anyio
import aiomqtt


async def handle(message):
    await anyio.sleep(5)  # Simulate some I/O-bound work
    print(message.payload)


async def main():
    async with aiomqtt.Client("test.mosquitto.org") as client:
        await client.subscribe("temperature/#")
        # Use a task group to manage and await all tasks
        async with anyio.create_task_group() as tg:
            async for message in client.messages:
                tg.create_task(handle(message))  # Spawn new coroutine


anyio.run(main)
```

**Comment by jimemo** [comment]
Thanks for your reponse and suggestion. When I run this in Python3.9.2 I get the following error. I have the latest versions of anyio and aiomqtt from Pip.

`Exception has occurred: ExceptionGroup
unhandled errors in a TaskGroup (1 sub-exception)
  File "/opt/bitnami/apache2/htdocs/py/mqtt_mysql/aio_test.py", line 15, in main
    tg.create_task(handle(message))  # Spawn new coroutine
AttributeError: 'TaskGroup' object has no attribute 'create_task'

During handling of the above exception, another exception occurred:

  File "/opt/bitnami/apache2/htdocs/py/mqtt_mysql/aio_test.py", line 15, in main
    tg.create_task(handle(message))  # Spawn new coroutine
  File "/opt/bitnami/apache2/htdocs/py/mqtt_mysql/aio_test.py", line 17, in <module>
    anyio.run(main)`

**Comment by frederikaalund** [comment]
T equivalent method of `asyncio.TaskGroup.create_task` is `anyio.TaskGroup.start_soon`. Try with `start_soon` instead of `create_task`. :)

**Comment by jimemo** [comment]
I can see that the sleep_until function is found, however I now get this error:
```
Exception has occurred: ExceptionGroup
unhandled errors in a TaskGroup (1 sub-exception)
  File "/opt/bitnami/apache2/htdocs/py/mqtt_mysql/aio_test.py", line 16, in main
    tg.start_soon(handle(message))  # Spawn new coroutine
TypeError: 'coroutine' object is not callable

During handling of the above exception, another exception occurred:

  File "/opt/bitnami/apache2/htdocs/py/mqtt_mysql/aio_test.py", line 16, in main
    tg.start_soon(handle(message))  # Spawn new coroutine
  File "/opt/bitnami/apache2/htdocs/py/mqtt_mysql/aio_test.py", line 19, in <module>
    anyio.run(main)
exceptiongroup.ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
```

**Comment by frederikaalund** [comment]
This is issue seems more about the use of asyncio/anyio than it is about the use of aiomqtt. Try to consult the anyio documentation/discussions for reference (or stackoverflow and the like). :+1:

In any case, last bit of help from me here :smile:: Try with `tg.start_soon(handle, message)`.

**Comment by HMaker** [comment]
> `asyncio.TaskGroup` is only available in Python 3.11+. You could use [AnyIO](https://anyio.readthedocs.io/en/stable/tasks.html). AnyIO use asyncio under the hood and makes using async in Python much easier.

Then the minimum python version should bet set to 3.11 in the pyproject.toml file, no?
```toml
[tool.poetry.dependencies]
python = "^3.11"
```
The pypi package classifiers should be also update to reflect that.

**Comment by empicano** [comment]
> Then the minimum python version should bet set to 3.11 in the `pyproject.toml` file, no?

The minimum version is correct, aiomqtt works from Python 3.8 upwards. However, the examples in the documentation sometimes use features from newer Python versions (like `asyncio.TaskGroup`).
